sunbeam_migrate/handlers/neutron/floating_ip.py

```diff
@@ -0,0 +1,171 @@
+
+        associated_resources = []
+        floating_network_id = source_fip.floating_network_id
+        if floating_network_id:
+            associated_resources.append(("network", floating_network_id))
+
+        subnet_ids = set()
+        if getattr(source_fip, "subnet_id", None):
+            subnet_ids.add(source_fip.subnet_id)
+
+        floating_ip = getattr(source_fip, "floating_ip_address", None)
+        if floating_ip:
+            try:
+                floating_ip_addr = ipaddress.ip_address(floating_ip)
+            except ValueError:
+                floating_ip_addr = None
+
+        if floating_ip_addr:
+            for subnet in self._source_session.network.subnets(
+                network_id=floating_network_id
+            ):
+                cidr = getattr(subnet, "cidr", None)
+                if not cidr:
+                    continue
+                try:
+                    network = ipaddress.ip_network(cidr, strict=False)
+                except ValueError:
+                    continue
+
+                if floating_ip_addr in network:
+                    # The Floating IP might not have a subnet_id set,
+                    # but requires a subnet based on the IP address.
+                    subnet_ids.add(subnet.id)
+                    break
+
+        for subnet_id in sorted(subnet_ids):
+            associated_resources.append(("subnet", subnet_id))
+        return associated_resources
+
+    def get_member_resource_types(self) -> list[str]:
+        """Get a list of member (contained) resource types.
+
+        The migrations can cascade to contained resources.
+        """
+        return []
+
+    def perform_individual_migration(
+        self,
+        resource_id: str,
+        migrated_associated_resources: list[tuple[str, str, str]],
+    ) -> str:
+        """Migrate the specified floating IP.
+
+        :param resource_id: the resource to be migrated
+        :param migrated_associated_resources: a list of tuples describing
+            associated resources that have already been migrated.
+            Format: (resource_type, source_id, destination_id)
+
+        Return the resulting resource id.
+        """
+        source_fip = self._source_session.network.get_ip(resource_id)
+        if not source_fip:
+            raise exception.NotFound(f"Floating IP not found: {resource_id}")
+
+        destination_network_id = self._get_associated_resource_destination_id(
+            "network",
+            source_fip.floating_network_id,
+            migrated_associated_resources,
+        )
+
+        dest_subnet_id = None
+        if getattr(source_fip, "subnet_id", None):
+            try:
+                dest_subnet_id = self._get_associated_resource_destination_id(
+                    "subnet",
+                    source_fip.subnet_id,
+                    migrated_associated_resources,
+                )
+            except exception.NotFound:
+                dest_subnet_id = None
```

Comment:
I don't think we should catch this exception. If there was an associated subnet that we didn't migrate, then this exception should be propagated.